CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+
+For a comprehensive guide on how to write small PRs, see this [guide](https://github.com/google/eng-practices/blob/master/review/developer/small-cls.md)
+
+## Contributor License Agreement (CLA)
+
+In order to accept your pull request, we need you to submit a CLA. You only need to do this once. Simply reply to your first pull request with `I have read the CLA Document and I hereby sign the CLA`. Your signature will be recorded automatically in the `cla` branch.
+
+## Code Reviews
+
+A code review is a process where someone other than the author(s) of a piece of code examines that code. Code committed to the codebase is both the responsibility of the author and the reviewer. Code reviews should look at:
+
+-   Design: Is the code well-designed and appropriate for your system?
+-   Functionality: Does the code behave as the author likely intended? Is the way the code behaves good for its users?
+-   Complexity: Could the code be made simpler? Would another developer be able to easily understand and use this code when they come across it in the future? Beware of over-engineering. The code should solve problems that need to be solved _now_, and not problems that the code author speculates _might_ need to be solved in the future.
+-   Tests: Does the code have correct and well-designed automated tests?
+-   Naming: Did the developer choose clear names for variables, classes, methods, etc.?
+-   Comments: Are the comments clear and useful? Note that comments are most useful when they explain _why_ the code exists.
+-   Style: Does the code follow our style guides? Note, in most cases, style nits should be avoided and be enforced entirely by automated tooling. However some stylistic decisions can be discussed if it impacts readability and complexity.
+-   Documentation: Did the developer also update relevant documentation?
+
+All of the above are grounds for a reviewer to request changes in a PR. Consensus should be reached to the best of the abilities of the author(s) and reviewer. However, if consensus cannot be reached between the two parties, the review should be escalated to the technical lead.
+
+### Code Review Conduct
+
+-   Be kind.
+-   Explain your reasoning.
+-   Balance giving explicit directions with just pointing out problems and letting the developer decide.
+-   Encourage developers to simplify code or add code comments instead of just explaining the complexity to you.
+
+Remember that you are both on the same team and are working towards the same goal. The goal is not to make the code perfect, but to make it better. Make sure your review comments are constructive and actionable. Treat the code contribution as a communal artifact under construction rather than a personal creation under evaluation. For a comprehensive guide on how to provide effective PR feedback, see this [guide](https://google.github.io/eng-practices/review/reviewer/comments.html).
```

Comment:
[source](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Aspects%20of%20software,of%20the%20system.)

[source](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free%20to%20leave%20comments%20expressing%20that%20something%20could%20be%20better%2C%20but%20if%20it%E2%80%99s%20not%20very%20important%2C%20prefix%20it%20with%20something%20like%20%E2%80%9CNit%3A%20%E2%80%9C%20to%20let%20the%20author%20know%20that%20it%E2%80%99s%20just%20a%20point%20of%20polish%20that%20they%20could%20choose%20to%20ignore.)
```suggestion
Remember that you are both on the same team and are working towards the same goal. The goal is not to make the code perfect, but to make it better. Make sure your review comments are constructive and actionable. Treat the code contribution as a communal artifact under construction rather than a personal creation under evaluation. 

Aspects of software design are almost never a pure style issue or just a personal preference. They are based on underlying principles and should be weighed on those principles, not simply by personal opinion. Sometimes there are a few valid options. If the author can demonstrate (either through data or based on solid engineering principles) that several approaches are equally valid, then the reviewer should accept the preference of the author. Otherwise the choice is dictated by standard principles of software design. If no other rule applies, then the reviewer may ask the author to be consistent with what is in the current codebase, as long as that doesn’t worsen the overall code health of the system.

Reviewers should _always_ feel free to leave comments expressing that something could be better, but if it’s not very important, prefix it with something like “Nit: “ to let the author know that it’s just a point of polish that they could choose to ignore.

For a comprehensive guide on how to provide effective PR feedback, see this [guide](https://google.github.io/eng-practices/review/reviewer/comments.html).
```

Reply:
While I don't disagree with some of the sentiment and source material you are highlighting I don't want to spell out every nuanced aspect of PRs - largely because I don't want this to be an artifact that is doctrine. I think this is why I linked out to the guidelines in places where nuance can be inferred but am going to not add to this document as is because it's brevity is more important right now than it's completeness.  I'm gonna optemize for simplicity to start and can extend it as needed.

Reply:
ok. sounds good. thanks